trezor/trezor-suite
# Feat/splashscreen tor

This PR creates a splash screen when Tor is loading that allows the user to see the real progress of Tor starting and it gives the opportunity to disable it and retry when there is any error.

https://github.com/trezor/trezor-suite/issues/5805

* 9d19dfe4c0a43f5d5a971ba3e34041e89c2b4b31 - Wait for Tor to be ready before loading the app
* 335fb5d6b000cbcdd5a730f1a33f3e3cbe9335d0 - Listen/logs/emit event of the real Tor progress so we can have a progress bar that displays the real situation and as well not user focus but nice for developer, it logs the summary messages from Tor so the dev knows what is happening.
* 09dfdd625b052be9953133595db12865f4122969 - Allows Tor to be stopped when it is in the process of starting, this is necessary for the "Disable Tor" functionality
* 1a61bbb3b01a6935731967a9284a8eeb59b1a474 - Splash screen with progress bar and options to disable as well as to try again to start Tor when it fails
* 5b4e8bb7ebe3a3ce04b547754c7b8a7a77526049 - Re-organizes Tor ControlPort closing.

Notes:
* In TorLoadingScreen there are hard-coded non translated messages because the Intl Provider is not loaded yet, it makes more complex the use of translation at this point so we can have a follow-up with that topic.

## How to test it
There are few scenarios that can be tested here
1. Starting suite desktop when Tor was not enable in the settings
    * expected => it load the app and nothing from the Tor splash screen should be visible 
2. Starting suite desktop when Tor was enable in the settings
    * expected => it displays a window with a progress bar that goes to 100% and then it load the normal suite app
3. Starting suite desktop when Tor was enable in settings and disable Tor before it completes the progress
    * expected => it display a progress bar that goes to 100% 
    * expected =>if you go to setting you should see Tor disable
    * expected => Tor process should be properly closed
4. Starting suite desktop when Tor was enable in settings and make it fail for example disconnecting the internet from your computer
    * expected => it should display a red progress bar and allow the user to "Disable Tor" and "Try again"
    * expected => when "Try again" is pressed it should start new Tor process and if now you connect your computer to the internet it should complete the progress bar and load the app normally.
